### ISSUE TITLE
target: no valid voxel when computing loss

### ISSUE THREAD

**Issue by MarklearnML** [zju3dv/NeuralRecon#116]
Hi,I got same troubles when training, just as the follow. Why I will get this? I guess it come from the full_tsdf_layer.npz.  Because I find that my full_tsdf_layer.npz's values is all 1. But why?

请原谅我英语不太好，作者您好，我训练的时候出现了下面的问题 说是没有target的值 我猜是generate_gt 生成的tsdf有点问题
因为我发现all_full_9里的full_tsdf_layer里面的值全为1 我不太清楚这个正不正常 如果不正常 那么又是什么原因呢 我觉得应该是fusion.py里的出现了点问题 但我并没有修改里面的源码 作者看到了能帮忙解答一下吗 非常感激!!!

2022-10-14 19:52:10.801 | WARNING  | models.neucon_network:compute_loss:242 - target: no valid voxel when computing loss
2022-10-14 19:52:11.161 | WARNING  | models.neucon_network:compute_loss:242 - target: no valid voxel when computing loss
2022-10-14 19:52:11.555 | WARNING  | models.neucon_network:compute_loss:242 - target: no valid voxel when computing loss
2022-10-14 19:52:12.057 | INFO     | __main__:train:212 - Epoch 8/991, Iter 0/1632, train loss = 0.000, time = 19.652
2022-10-14 19:52:12.429 | WARNING  | models.neucon_network:compute_loss:242 - target: no valid voxel when computing loss
2022-10-14 19:52:12.531 | WARNING  | models.neucon_network:compute_loss:242 - target: no valid voxel when computing loss
2022-10-14 19:52:12.603 | WARNING  | models.neucon_network:compute_loss:242 - target: no valid voxel when computing loss
2022-10-14 19:52:13.011 | INFO     | __main__:train:212 - Epoch 8/991, Iter 1/1632, train loss = 0.000, time = 0.765
2022-10-14 19:52:13.202 | WARNING  | models.neucon_network:compute_loss:242 - target: no valid voxel when computing loss
2022-10-14 19:52:13.355 | WARNING  | models.neucon_network:compute_loss:242 - target: no valid voxel when computing loss
2022-10-14 19:52:13.500 | WARNING  | models.neucon_network:compute_loss:242 - target: no valid voxel when computing loss
2022-10-14 19:52:13.888 | INFO     | __main__:train:212 - Epoch 8/991, Iter 2/1632, train loss = 0.000, time = 0.867
2022-10-14 19:52:14.118 | WARNING  | models.neucon_network:compute_loss:242 - target: no valid voxel when computing loss
2022-10-14 19:52:14.211 | WARNING  | models.neucon_network:compute_loss:242 - target: no valid voxel when computing loss
2022-10-14 19:52:14.280 | WARNING  | models.neucon_network:compute_loss:242 - target: no valid voxel when computing loss
2022-10-14 19:52:14.678 | INFO     | __main__:train:212 - Epoch 8/991, Iter 3/1632, train loss = 0.000, time = 0.737
2022-10-14 19:52:14.880 | WARNING  | models.neucon_network:compute_loss:242 - target: no valid voxel when computing loss
2022-10-14 19:52:15.005 | WARNING  | models.neucon_network:compute_loss:242 - target: no valid voxel when computing loss
2022-10-14 19:52:15.073 | WARNING  | models.neucon_network:compute_loss:242 - target: no valid voxel when computing loss
2022-10-14 19:52:15.453 | INFO     | __main__:train:212 - Epoch 8/991, Iter 4/1632, train loss = 0.000, time = 0.759
2022-10-14 19:52:15.640 | WARNING  | models.neucon_network:compute_loss:242 - target: no valid voxel when computing loss
2022-10-14 19:52:15.730 | WARNING  | models.neucon_network:compute_loss:242 - target: no valid voxel when computing loss
2022-10-14 19:52:15.791 | WARNING  | models.neucon_network:compute_loss:242 - target: no valid voxel when computing loss
2022-10-14 19:52:16.166 | INFO     | __main__:train:212 - Epoch 8/991, Iter 5/1632, train loss = 0.000, time = 0.692
2022-10-14 19:52:16.338 | WARNING  | models.neucon_network:compute_loss:242 - target: no valid voxel when computing loss
2022-10-14 19:52:16.432 | WARNING  | models.neucon_network:compute_loss:242 - target: no valid voxel when computing loss
2022-10-14 19:52:16.514 | WARNING  | models.neucon_network:compute_loss:242 - target: no valid voxel when computing loss
2022-10-14 19:52:16.885 | INFO     | __main__:train:212 - Epoch 8/991, Iter 6/1632, train loss = 0.000, time = 0.712
2022-10-14 19:52:17.059 | WARNING  | models.neucon_network:compute_loss:242 - target: no valid voxel when computing loss
2022-10-14 19:52:17.155 | WARNING  | models.neucon_network:compute_loss:242 - target: no valid voxel when computing loss
2022-10-14 19:52:17.220 | WARNING  | models.neucon_network:compute_loss:242 - target: no valid voxel when computing loss
2022-10-14 19:52:17.601 | INFO     | __main__:train:212 - Epoch 8/991, Iter 7/1632, train loss = 0.000, time = 0.708
2022-10-14 19:52:17.782 | WARNING  | models.neucon_network:compute_loss:242 - target: no valid voxel when computing loss
2022-10-14 19:52:17.892 | WARNING  | models.neucon_network:compute_loss:242 - target: no valid voxel when computing loss
2022-10-14 19:52:18.038 | WARNING  | models.neucon_network:compute_loss:242 - target: no valid voxel when computing loss
2022-10-14 19:52:18.467 | INFO     | __main__:train:212 - Epoch 8/991, Iter 8/1632, train loss = 0.000, time = 0.856
2022-10-14 19:52:18.847 | WARNING  | models.neucon_network:compute_loss:242 - target: no valid voxel when computing loss
2022-10-14 19:52:19.085 | WARNING  | models.neucon_network:compute_loss:242 - target: no valid voxel when computing loss
2022-10-14 19:52:19.218 | WARNING  | models.neucon_network:compute_loss:242 - target: no valid voxel when computing loss

**Comment by MarklearnML** [comment]
I solve the problem!

**Comment by ktiays** [comment]
I also met the same problem. How did you solve it?

**Comment by ktiays** [comment]
> I solve the problem!

I also met the same problem. How did you solve it?

**Comment by everythoughthelps** [comment]
> 

I also met the same problem, how did you solve it?

**Comment by everythoughthelps** [comment]
> > I solve the problem!
> 
> I also met the same problem. How did you solve it?

did you solve the problem?

**Comment by MarklearnML** [comment]
> > 
> 
> I also met the same problem, how did you solve it?
Are you using the complete ScanNet dataset or the 5G mini dataset? It seems that the author's source code is based on the complete dataset. If you're using the 5G dataset, you may encounter errors and need to modify the file names.

**Comment by MarklearnML** [comment]
> > 
> 
> I also met the same problem, how did you solve it?

Are you using the complete ScanNet dataset or the 5G mini dataset? It seems that the author's source code is based on the complete dataset. If you're using the 5G dataset, you may encounter errors and need to modify the file names.

**Comment by everythoughthelps** [comment]
谢了老哥，我用的那个袖珍数据集，要改好多bug



Sent&nbsp;from&nbsp;my&nbsp;iPhone


------------------ Original ------------------
From: GuoyuHuang ***@***.***&gt;
Date: Tue,Aug 29,2023 4:15 PM
To: zju3dv/NeuralRecon ***@***.***&gt;
Cc: Pan Meng ***@***.***&gt;, Comment ***@***.***&gt;
Subject: Re: [zju3dv/NeuralRecon] target: no valid voxel when computing loss (Issue #116)





    
I also met the same problem, how did you solve it?
 Are you using the complete ScanNet dataset or the 5G mini dataset? It seems that the author's source code is based on the complete dataset. If you're using the 5G dataset, you may encounter errors and need to modify the file names.
